soql-standalone-parser/src/main/scala/com/socrata/soql/ast/Select.scala

```diff
@@ -42,11 +42,23 @@ case class JoinTable(tableName: TableName) extends JoinSelect {
 }
 case class JoinQuery(selects: BinaryTree[Select], definiteAlias: String) extends JoinSelect {
   val alias = Some(definiteAlias)
-  override def toString = "(" + Select.toString(selects) + ") AS @" + TableName.removeValidPrefix(definiteAlias)
+
+  override def toString = {
+    if (selects.inParen) {
+      Select.toString(selects) + " AS @" + TableName.removeValidPrefix(definiteAlias)
+    } else {
+      "(" + Select.toString(selects) + ") AS @" + TableName.removeValidPrefix(definiteAlias)
+    }
+  }
+
   def replaceHoles(f: Hole => Expression): JoinQuery =
     copy(Select.walkTreeReplacingHoles(selects, f))
-  def rewriteJoinFuncs(f: Map[TableName, UDF], aliasProvider: AliasProvider): JoinQuery =
-    copy(selects = Select.rewriteJoinFuncs(selects, f)(aliasProvider))
+
+  def rewriteJoinFuncs(f: Map[TableName, UDF], aliasProvider: AliasProvider): JoinQuery = {
+    val rewritten = Select.rewriteJoinFuncs(selects, f)(aliasProvider)
+    copy(selects = rewritten.wrapInParen)
```

Comment:
Most changes indeed just preserve parens, but we're also consciously adding them to join functions sometime? When are they rewritten like this?

Reply:
i think we always attempt to rewrite joins.  They are actually rewritten when they are UDFs.  Parens are always safe and necessary for query joins.